starters/vue3-xstate-css/src/machines/counterMachine.ts

```diff
@@ -0,0 +1,32 @@
+import { createMachine, assign } from 'xstate';
+
+export const counterMachine = createMachine(
+	{
+		id: 'Counter',
+		predictableActionArguments: true,
+		schema: {
+			context: {} as { count: number },
+			events: {} as { type: 'INC' } | { type: 'DEC' } | { type: 'RESET' },
+		},
+		initial: 'active',
+		context: {
+			count: 0,
+		},
+		states: {
+			active: {
+				on: {
+					INC: { actions: 'increment' },
+					DEC: { actions: 'decrement' },
+					RESET: { actions: 'reset' },
+				},
+			},
+		},
+	},
+	{
+		actions: {
+			increment: assign({ count: (context) => context.count + 1 }),
+			decrement: assign({ count: (context) => context.count - 1 }),
+			reset: assign({ count: (context) => (context.count = 0) }),
+		},
+	}
```

Comment:
When I try to re-situate it like this, I'm getting type errors for `on`, and nothing I'm finding in the docs is helping to alleviate that.

But also, from this section of the docs on actions, they recommend declaring the actions in the config object that's passed as the second object, which is how I have it here already. So despite slightly conflicting info, I think the way it currently is might actually be a better practice overall.
<img width="758" alt="Screen Shot 2023-05-05 at 6 55 19 PM" src="https://user-images.githubusercontent.com/16214572/236587833-859272dc-629d-4777-a2ca-d9ecfc1091f0.png">
https://xstate.js.org/docs/guides/actions.html#api

The condition isn't set because in the starter specifications, it says it should be able to work in the negative direction as well: https://docs.google.com/document/d/1iEXK6BFgu5HZ4dfU7pfv80zLOAd86T8AzstCMEYTUXA/edit#heading=h.o7at9s7kpri4